pom.xml

```diff
@@ -693,6 +670,145 @@
             </properties>
         </profile>
 
+
+        <profile>
+            <id>java9plus</id>
+            <activation>
+                <jdk>[9,)</jdk>
+            </activation>
+            <build>
+                <plugins>
+                    <!-- Don't compile module-info.java, see java 9 profile -->
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-compiler-plugin</artifactId>
+                        <executions>
+                            <execution>
+                                <id>default-compile</id>
+                                <phase>none</phase>
+                            </execution>
+                            <execution>
+                                <id>default-testCompile</id>
+                                <configuration>
+                                    <release>9</release>
+                                    <encoding>UTF-8</encoding>
+                                    <forceJavacCompilerUse>true</forceJavacCompilerUse>
+                                    <compilerArgument>-proc:none</compilerArgument>
+                                    <fork>false</fork>
+                                    <compilerArgument>-proc:none</compilerArgument>
+                                </configuration>
+                            </execution>
+                            <execution>
+                                <id>java-9-compile</id>
+                                <goals>
+                                    <goal>compile</goal>
+                                </goals>
+                                <configuration>
+                                    <release>9</release>
+                                    <!-- no excludes: compile everything to ensure module-info contains right entries -->
```

Comment:
What does this mean? There are excludes below.

Reply:
Removing.